mp/src/game/client/momentum/ui/controls/UserComponent.cpp

```diff
@@ -134,4 +129,12 @@ void UserComponent::OnMouseReleased(MouseCode code)
     {
         PostActionSignal(new KeyValues("UserComponentClicked"));
     }
-}
\ No newline at end of file
+}
+
+void UserComponent::OnPersonaStateChange(PersonaStateChange_t *pParams)
+{
+    if (SteamUser()->GetSteamID().ConvertToUint64() == pParams->m_ulSteamID)
```

Comment:
This should be comparing if the user component's user is the one that changed. Furthermore, it seems like from [the documentation on this](https://partner.steamgames.com/doc/api/ISteamFriends#PersonaStateChange_t), you'll want to verify if it's just the name that changed using `pParams->m_nChangeFlags` and bitwising the [flags found here](https://partner.steamgames.com/doc/api/ISteamFriends#EPersonaChange) for the name change and maybe even that "name first set" flag too.

Reply:
Not sure what you mean by the first bit. I changed it to use user data for now.

Reply:
I mean that this component may be used to represent other players in the future. Right now it's only used in the main menu but it's a generic component for that reason, otherwise it would have been called like "LocalUserComponent".

I think adding a new private CSteamID variable and tracking the ID coming from SetUser functions in it, then using that here is the approach to take.